elastic/kibana
# [Fleet] fixed bug in output secret diff logic

## Summary

Related to https://github.com/elastic/kibana/issues/104986

Found a bug in `diffOutputSecretPaths` where output secret was deleted if updating an output without change of service_token. Added unit tests to cover the logic.

Steps to verify:
- enable feature flags: `xpack.fleet.enableExperimental: ['remoteESOutput', 'outputSecretsStorage']`
- create a remote es output with a service_token
- check that the service_token is stored as secret in `.fleet-secrets`
- update host in remote es output
- verify that the secret is not deleted


### Checklist
